modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/Walker.java

```diff
@@ -918,7 +918,7 @@ public ANode visitVariable(VariableContext ctx) {
 
     @Override
     public ANode visitCalllocal(CalllocalContext ctx) {
-        String name = ctx.ID().getText();
+        String name = ctx.ID() == null ? ctx.DOLLAR().getText() : ctx.ID().getText();
```

Comment:
Will name be `$` in this case?

Reply:
In this case, yes. But if we add additional elements unintentionally this could be null and would lead to an early NPE in a test case, so I think it's worth it to leave as is.